promql/functions.go

```diff
@@ -938,6 +936,28 @@ var FunctionCalls = map[string]FunctionCall{
 	"year":               funcYear,
 }
 
+// AtModifierUnsafeFunctions are the functions whose result
+// can vary if evaluation time is changed when the arguments are
+// step invariant. It also includes functions that use the timestamp
+// of the passed argument to calculate a result since that can
+// also change with change in eval time.
+var AtModifierUnsafeFunctions = map[string]struct{}{
+	// Step invariant functions.
+	"days_in_month": {}, "day_of_month": {}, "day_of_week": {},
+	"hour": {}, "minute": {}, "month": {}, "year": {},
+	"predict_linear": {}, "time": {},
+	// Uses timestamp of the argument for the result,
+	// hence unsafe to use with @ modifier.
+	"timestamp": {},
+}
+
+// IsFunctionAtModifierSafe returns true if the function
+// is safe to wrap with step invariant node in case of @ modifier.
+func IsFunctionAtModifierSafe(e *parser.Call) bool {
```

Comment:
Quite a shallow function to me. We can totally  inline this with just one liner. Not a blocker though.

Reply:
Right, this is no longer required since now we have `AtModifierUnsafeFunctions`